sticker/src/tensorflow/dataset.rs

```diff
@@ -151,3 +146,65 @@ where
         Some(Ok(builder))
     }
 }
+
+/// Trait providing adapters for `conllx::io::Sentences`.
+pub trait SentenceIter: Sized {
+    fn filter_by_len(self, max_len: usize) -> LengthFilter<Self>;
+}
+
+impl<I> SentenceIter for I
+where
+    I: Iterator<Item = Result<Sentence, Error>>,
+{
+    fn filter_by_len(self, max_len: usize) -> LengthFilter<Self> {
+        LengthFilter {
+            inner: self,
+            max_len,
+        }
+    }
+}
+
+/// An Iterator adapter filtering sentences by maximum length.
+pub struct LengthFilter<I> {
+    inner: I,
+    max_len: usize,
+}
+
+impl<I> Iterator for LengthFilter<I>
+where
+    I: Iterator<Item = Result<Sentence, Error>>,
+{
+    type Item = Result<Sentence, Error>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        while let Some(sent) = self.inner.next() {
+            let len = sent.as_ref().map(|s| s.len()).unwrap_or(0);
+            if len > self.max_len {
+                continue;
+            }
+            return Some(sent);
+        }
+        None
+    }
+}
+
+/// Returns an `Iterator` over `Result<Sentence, Error>`.
+///
+/// Depending on the parameters the returned iterator filters
+/// sentences by their lengths or returns the sentences in
+/// sequence without filtering them.
+///
+/// If `max_len` == `usize::MAX`, no filtering is performed.
```

Comment:
I know that I am the guilty party here, but interface-wise, it would be nicer to take an `Option`, so that `None` can be used for no filtering. This can be a separate PR.